### DISCUSSION TITLE
Add support for singed url upload on client

### DISCUSSION THREAD

**Discussion by rawnly** [supabase/supabase#12958] [Feature Requests]
Recently came across https://github.com/supabase/storage-api/pull/282 , but seems like is not supported by `supabase-js` yet, please correct me if I'm wrong.

It would be nice to have a way to upload via the supbase-js client too.

**Comment by GaryAustin1** [comment]
I don't think it is officially released yet.  So use at your own risk.

**Reply by rawnly** [reply]
Do you mean that even if it's released on GH (the docker image is out too) it might not be on supabase?
https://github.com/supabase/storage-api/releases/tag/v0.30.0

**Reply by GaryAustin1** [reply]
It all depends on the priority for updating all the instances storage-api code.  Sometimes if there is a bug Supabase seems to role out quickly other times takes a week or more.  For instance PostgREST 11 was announced a month ago for the local image, but is not on the hosted instances yet.

Until something is announced like in the Feb new features notices or is in docs, there is no guarantee it is rolled out, or even fully functional.   It could just be part of the solution.  I would say use at your own risk until then.
![image](https://user-images.githubusercontent.com/54564956/224511003-7a15aa2b-58b3-4754-a239-bdc8f755f56c.png)

**Reply by rawnly** [reply]
Thanks for the clear explanation!

**Reply by GaryAustin1** [reply]
If you are on a free instance, you can pause/restore to pull down whatever Supabase considers the latest code (which may or may not contain the code you showed) and try it and see.
